kafka-plugins-0.10/src/test/java/io/cdap/plugin/KafkaStreamingSourceTest.java

```diff
@@ -264,7 +264,7 @@ public void testKafkaStreamingSourceFromLastOffset() throws Exception {
     sparkManager.waitForStatus(true, 10, 1);
 
     // for this test we have to wait for pipeline to be in running state before sending data
-    sparkManager.startAndWaitForRun(ProgramRunStatus.RUNNING, 20, TimeUnit.SECONDS);
+    sparkManager.waitForRun(ProgramRunStatus.RUNNING, 20, TimeUnit.SECONDS);
```

Comment:
Were we starting the pipeline twice before your fix (first started in line 263)?

Reply:
We do `sparkManager.start()` in line 182 and `sparkManager.startAndWaitForRun` in here. startAndWaitForRun explicitly says it should not be called with another run in progress

Reply:
Sorry, not 182, line  263